pkg/cmd/pulumi/packagecmd/project_sdk_link.go

```diff
@@ -716,20 +725,12 @@ func ProviderFromSource(packageSource string) (plugin.Provider, error) {
 		return nil, err
 	}
 
-	descriptor := workspace.PackageDescriptor{
-		PluginSpec: workspace.PluginSpec{
-			Kind: apitype.ResourcePlugin,
-			Name: packageSource,
-		},
+	pluginSpec, err := workspace.NewPluginSpec(packageSource, apitype.ResourcePlugin, nil, "", nil)
+	if err != nil {
+		return nil, err
 	}
-
-	if s := strings.SplitN(packageSource, "@", 2); len(s) == 2 {
-		descriptor.Name = s[0]
-		v, err := semver.ParseTolerant(s[1])
-		if err != nil {
-			return nil, fmt.Errorf("VERSION must be valid semver: %w", err)
-		}
-		descriptor.Version = &v
+	descriptor := workspace.PackageDescriptor{
+		PluginSpec: pluginSpec,
```

Comment:
Can't say I understand the implications of version and download URL changes in this file. Could you please comment on those?

Reply:
There's no changes for non-git URLs (other than `github.com/bla/bla` would now be treated as a remote URL, not a local plugin).  The main change here is that we now support remote plugins from git sources.